lib/services/project-templates-service.ts

```diff
@@ -1,13 +1,99 @@
 ///<reference path="../.d.ts"/>
 "use strict";
+import * as path from "path";
+import * as temp from "temp";
+import * as constants from "../constants";
+temp.track();
 
 export class ProjectTemplatesService implements IProjectTemplatesService {
-	private static NPM_DEFAULT_TEMPLATE_NAME = "tns-template-hello-world";
+	private static RESERVED_TEMPLATE_NAMES: IStringDictionary = {
```

Comment:
In fact the default templates are case-insensitive:
https://github.com/NativeScript/nativescript-cli/pull/1218/files#diff-123e2a8a4e8daddf9ac8a60b8c79546dR29

In case the lowercased value is not one of the reserved template names, we'll call `npm install` with original value specified by user. We do not want to set toLowerCase() there as the specified value might be case sensitive URL.